optimum/exporters/executorch/recipes/cuda.py

```diff
@@ -0,0 +1,127 @@
+    Returns:
+        Dict[str, ExecutorchProgram]:
+            A map of exported and optimized program for ExecuTorch.
+            For encoder-decoder models or multimodal models, it may generate multiple programs.
+    """
+    # Import here to avoid version conflicts.
+    from torch._inductor.decomposition import conv1d_to_conv2d
+
+    from executorch.backends.cuda.cuda_backend import CudaBackend
+    from executorch.backends.cuda.cuda_partitioner import CudaPartitioner
+
+    def _lower_to_executorch(
+        exported_programs: Dict[str, ExportedProgram],
+        metadata=None,
+    ) -> Dict[str, ExecutorchProgram]:
+        logging.debug(f"\nExported program: {exported_programs}")
+
+        # If just one exported program, the method name in the .pte for it should be "forward".
+        if len(exported_programs) == 1:
+            exported_programs = {"forward": next(iter(exported_programs.values()))}
+
+        # CUDA backend compile spec with method name.
+        partitioners = {key: [CudaPartitioner([CudaBackend.generate_method_name_compile_spec(key)])] for key in exported_programs.keys()}
+        # Add decompositions for triton to generate kernels.
+        for key, ep in exported_programs.items():
+            exported_programs[key] = ep.run_decompositions({
+              aten.conv1d.default: conv1d_to_conv2d,
+            })
+        with torch.nn.attention.sdpa_kernel([SDPBackend.MATH]):
+          et_prog = to_edge_transform_and_lower(
+              exported_programs,
+              partitioner=partitioners,
+              compile_config=EdgeCompileConfig(
+                  _check_ir_validity=False,
+                  _skip_dim_order=True,
+              ),
+              constant_methods=metadata,
+              transform_passes=[RemovePaddingIdxEmbeddingPass()],
+          )
+        print(et_prog.exported_program("text_decoder").graph_module.graph)
+        et_prog = et_prog.to_executorch()
+        pte_name = "model"
+        for method in et_prog.methods:
+            logging.debug(f"---------------------- Method: {method} ----------------------")
+            logging.debug(f"\nExecuTorch program for {pte_name}.pte: {et_prog.exported_program(method).graph_module}")
+            delegation_info = get_delegation_info(et_prog.exported_program(method).graph_module)
+            logging.debug(f"\nDelegation info Summary for {pte_name}.pte: {delegation_info.get_summary()}")
+            logging.debug(
+                f"\nDelegation info for {pte_name}.pte: {tabulate(delegation_info.get_operator_delegation_dataframe(), headers='keys', tablefmt='fancy_grid')}"
+            )
+        return {pte_name: et_prog}
+
+    # Decomposes SDPA since we don't have a flash attention kernel for it yet.
+    with torch.nn.attention.sdpa_kernel([SDPBackend.MATH]), torch.no_grad():
+        exported_progs = model.export()
+
+    if (
+        model.config._attn_implementation == "custom_sdpa"
```

Comment:
this is a generic recipe. does all supported transformer models have _attn_implementation? I dont suppose you are introducing this new behavior but just asking